### DISCUSSION TITLE
Video "Stretch to fit"

### DISCUSSION THREAD

**Discussion by BazzaCuda** [FactoryXCode/MfPack#54] [General]
Is there a way to implement "Stretch to Fit" like in Windows Media Player?

I want my video to fill the dimensions of the video TPanel. I will then adjust the window height to suit the dimensions of the video.

I'm trying to eliminate the black borders that MMP creates either across the top and bottom of the TPanel, or on the left and right sides.

Thanks.

**Comment by FactoryXCode** [comment]
Hi,
Of course you can do this (lookin to the video player samples).
However, you will get out of proportion frames or you wil loose right and left data from the frames being displayed.
The usual way to fit a video, is to keep it's original format and adjust the size of the window that displays it.
To get the video size call IMFMediaEngineEx.GetNativeVideoSize(aVideoWidth, aVideoHeight);
Then adjust your TPanel Height and Width to  (aVideoHeight + BorderWidth) and (aVideoWidth + BorderWidth).
Keep in mind that most video formats have their own dimensions, like common used [Video formats](https://esahubble.org/press/video_formats/).

Regards, Tony.
